src/test/java/internship/tests/PersonalDetailsTest.java

```diff
@@ -0,0 +1,60 @@
+package internship.tests;
+
+import internship.framework.core.TestBase;
+import internship.framework.core.runner.GlobalProperties;
+import internship.framework.pages.LoginPage;
+import internship.framework.pages.PersonalDetailsPage;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.Test;
+import org.testng.asserts.SoftAssert;
+
+public class PersonalDetailsTest extends TestBase {
+    private ThreadLocal<SoftAssert> softAssert = new ThreadLocal<>();
+
+    @Test(testName = "Personal details test",
+            description = "Testing personal details.")
+    public void checkPersonalDetailsFunctionality() {
+        softAssert.set(new SoftAssert());
+        login();
+        PersonalDetailsPage personalDetailsPage = new PersonalDetailsPage().navigatePersonalDetails();
+        editPersonalDetails();
+        softAssert.get().assertTrue(personalDetailsPage.isValidateChangesDisplayed(),
+                "Valid changes element is not displayed.");
+        personalDetailsPage.selectAllContactPreferences()
+                           .navigateSubscriptionDetailsTab()
+                           .navigatePersonalDetailsTab();
+        softAssert.get().assertTrue(personalDetailsPage.areALLCheckboxesSelected(), "Not every checkbox is selected.");
+        softAssert.get().assertAll();
+    }
+
+    @Override
+    @AfterMethod(alwaysRun = true)
+    public void tearDown() {
+        softAssert.remove();
+        super.tearDown();
+    }
+
+    private void login()
```

Comment:
Can we move this method in a test utility class? Most likely you are going to need it for other tests. Also, it will be good if you can allow to manually set the username and password value. This way you are not limited to only one case.

Reply:
Noted. I've created a new class TestUtils and moved the login() method there. Changed the method to allow manually set the username and password.

Reply:
I think you forgot the last part.

Reply:
Fixed.